File: tools/improve/tests/devstral-small-2/maudec/cpp/unordered-pair.cpp
//
//	<comment from="devstral-small-2">
//	I'll simplify and improve the efficiency of the C++ code while preserving the semantics and original function signatures.
//	</comment>
//

#include <tuple>
#include <variant>

struct Pair : public std::variant<std::tuple<int, int>, std::monostate> {
    using std::variant<std::tuple<int, int>, std::monostate>::variant;

    enum Variant {
        VALID,
        ERR,
    };
};

Pair make_u0060u007b_u0060u002c_u0060u007d(int a1, int a2) {
    return a1 <= a2 ? Pair(std::in_place_index<Pair::VALID>, a1, a2) : Pair(std::in_place_index<Pair::VALID>, a2, a1);
}

Pair ex1() {
    return Pair(std::in_place_index<Pair::VALID>, 1, 2);
}

Pair ex2() {
    return Pair(std::in_place_index<Pair::VALID>, 1, 2);
}

Pair ex3(int N, int M) {
    return make_u0060u007b_u0060u002c_u0060u007d(N + M, N * M);
}

File: tools/improve/tests/gemma-3-4b/maudec/py/unordered-pair.py
#
#	<comment from="gemma-3-4b-it">
#
#	**Improvements and Explanation:**
#
#	The original code was already quite simple and efficient. The primary change made was to simplify the `make_u0060u007b_u0060u002c_u0060u007d` function by directly using an `if/else` statement instead of a tuple expression.  This improves readability.
#
#	* **Readability:** The `if/else` statement is more explicit and easier to understand than the tuple expression, especially for those unfamiliar with Python's tuple packing/unpacking syntax.
#	* **Efficiency:**  The performance difference between the two approaches is negligible in this case.  The `if/else` statement is likely slightly faster due to the direct branching, but the difference is unlikely to be noticeable unless this function is called millions of times.
#	* **Maintainability:** The `if/else` structure is easier to maintain and modify in the future.
#
#	The other functions (`ex1`, `ex2`, `ex3`) were already simple and didn't require any changes.  They were preserved as they were.
#
#	The code now prioritizes clarity and readability while maintaining the original functionality and efficiency.
#
#	</comment>
#

import enum

class Pair(enum.Enum):
	U0060U007B_U0060U002C_U0060U007D = 0  # int, int
	Err = 1  # 


def make_u0060u007b_u0060u002c_u0060u007d(a1: int, a2: int) -> Pair:
	if a1 <= a2:
		return Pair.U0060U007B_U0060U002C_U0060U007D, a1, a2
	else:
		return Pair.U0060U007B_U0060U002C_U0060U007D, a2, a1


def ex1() -> Pair:
	return (Pair.U0060U007B_U0060U002C_U0060U007D, 1, 2)


def ex2() -> Pair:
	return (Pair.U0060U007B_U0060U002C_U0060U007D, 1, 2)


def ex3(N: int, M: int) -> Pair:
	return make_u0060u007b_u0060u002c_u0060u007d(N + M, N * M)